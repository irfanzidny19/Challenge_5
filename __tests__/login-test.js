import 'react-native';
import React from 'react';
import LoginPage from '../src/screens/Login';
import renderer from 'react-test-renderer';

describe('login',() => {
    test('Login Page Snapshot', () =>{
        const snap = renderer.create(<LoginPage/>).toJSON();
        expect(snap).toMatchSnapshot();
    });

    it('function and state test care',()=>{
        let HomeData= renderer.create(<LoginPage />).getInstance();

        HomeData.setPassword()

        expect(HomeData.Password).toEqual('')
    })

    
});

