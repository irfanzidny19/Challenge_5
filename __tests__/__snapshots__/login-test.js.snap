// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login Login Page Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#545454",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/logo.png",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "marginTop": -106.72,
        "padding": 37.5,
        "width": 675,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Login
    </Text>
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 70,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 16,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderStyle": "solid",
                "borderWidth": 2,
                "bottom": 0,
                "left": -3,
                "position": "absolute",
                "right": -3,
                "top": -3,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "left": "-100%",
                "paddingLeft": "50%",
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "translateY": 26,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": "200%",
              }
            }
          >
            <Text
              collapsable={false}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 16,
                  "includeFontPadding": false,
                  "lineHeight": 16,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                }
              }
            >
              Email
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              autoCapitalize="sentences"
              clearTextOnFocus={true}
              disableFullscreenUI={true}
              editable={true}
              keyboardType="email-address"
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onContentSizeChange={[Function]}
              onFocus={[Function]}
              selectionColor="rgb(0, 145, 234)"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "includeFontPadding": false,
                    "margin": 0,
                    "padding": 0,
                    "paddingTop": 0,
                    "textAlign": "left",
                    "textAlignVertical": "top",
                    "top": 2,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, .87)",
                    "fontSize": 16,
                    "height": 24,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "minHeight": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
          ]
        }
      >
        <Text
          collapsable={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(0, 0, 0, .38)",
              "flex": 1,
              "fontSize": 12,
              "lineHeight": 16,
              "opacity": 1,
              "paddingVertical": 2,
              "textAlign": "left",
            }
          }
        >
          Ex: dummail@gmail.com
        </Text>
      </View>
    </View>
    <TextInput
      label="Password"
      labelFontSize={18}
      lineWidth={2}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "borderBottomColor": "rgb(0,0,38)",
          "borderBottomWidth": 1,
          "marginBottom": 5,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          login
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
          }
        }
      >
        Don't have an account? 
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(0,145,234)",
              "textDecorationLine": "underline",
            }
          }
        >
          Register
        </Text>
      </View>
    </View>
  </View>
</View>
`;
